app/views/pages/item-detail/index.js

```diff
@@ -0,0 +1,228 @@
+  let bodyHeight = document.body.getBoundingClientRect().height;
+  let headerHeight = document.getElementsByClassName('product__header-menu')[0].getBoundingClientRect().height;
+
+  return bodyHeight - headerHeight;
+}
+
+var ItemDetail = React.createClass({
+
+  mixins: [State, ItemDetailMixin],
+
+  getInitialState() {
+    let product = ProductStore.getProduct(this.getParams().id);
+
+    return {
+      item: {},
+      product: product,
+      attachmentsPanel: false,
+      itemDetailHeight: initialItemDetailHeight(),
+      descriptionEditable: false
+    };
+  },
+
+  updateStripeHeight() {
+    let content = document.getElementsByClassName('content__wrapper')[0];
+    let height = content ? content.getBoundingClientRect().height : 0;
+
+    if (height > this.state.itemDetailHeight) {
+      this.setState({itemDetailHeight: height})
+    }
+  },
+
+  componentWillReceiveProps(nextProps) {
+    this.updateStripeHeight();
+
+    if (this.getParams().number != this.state.item.number) {
+      ItemActions.fetchItem(this.getParams().id, this.props.number);
+      ItemActions.fetchActivity(this.getParams().id, this.getParams().number);
+      this._onChange();
+    }
+  },
+
+  setItem(key, ev, value) {
+    let newVal = _.isUndefined(value) ? ev.currentTarget.value : value;
+
+    let item = _.cloneDeep(this.state.item);
+    item[key] = newVal;
+
+    this.setState({item: item});
+  },
+
+  setSubitem(subitemId, key, ev, value) {
+    let item = _.cloneDeep(this.state.item);
+    let subitem = _.findWhere(item.sub_items, {number: subitemId})
+    subitem[key] = value;
+
+    this.setState({item: item});
+  },
+
+  updateItem() {
+    let productId = this.getParams().id;
+    let itemId = this.getParams().number;
+
+    ProductActions.updateItem(productId, itemId, { description: this.state.item.description });
+  },
+
+  itemDetails() {
+    return (
+      <ItemDetails  members={this.props.members}
+                       type={this.state.item.type}
+                      title={this.state.item.title}
+                        who={this.state.item.who}
+                       what={this.state.item.what}
+                        why={this.state.item.why}
+                     number={this.state.item.number}
+                       tags={this.state.item.tags}
+                  createdAt={this.state.item.created_at}
+                  createdBy={this.state.item.created_by}
+                     status={this.state.item.status}
+                     score={this.state.item.score}
+                   assignee={this.state.item.assigned_to}
+                attachments={this.state.item.attachments}
+                   setItem={this.setItem}
+                   productTags={this.state.product.tags} />
+    )
+  },
+
+  itemDescription() {
+    return (
+      <div className="col-xs-12 col-sm-12 col-md-12 col-lg-9 section description">
+        <div className="col-lg-12">
+          {this.header('description')}
```

Comment:
This is an example where we're better off with a component instead of making our own abstraction:

```
<DetailHeader title="description" />
```
Making this a one-off component has the benefit of knowing the details of the abstraction since it follows the pattern used by everything else in React.

As a reviewer, figuring out what `this.header('description')` is doing requires looking for it in the source (oh wait, it's in another file) and then reading the method—whereas the component is completely unambiguous about what it does when you invoke it.